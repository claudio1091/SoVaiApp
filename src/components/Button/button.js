import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const buttonHeight = 50;
const buttonWidth = 137;

const TouchableContainer = styled.TouchableHighlight`
  border-radius: 10;
  height: ${buttonHeight};
  margin-vertical: 5;
  width: ${buttonWidth};
`;

const ButtonContainer = styled.View`
  align-items: 'center';
  background-color: ${props => (props.primary ? '#EE6C4D' : 'transparent')};
  border-color: '#EE6C4D';
  border-radius: 10;
  border-width: ${props => (props.primary ? 0 : 3)};
  flex-direction: 'column';
  height: ${buttonHeight};
  justify-content: 'center';
  width: ${buttonWidth};
`;

const ButtonText = styled.Text`
  color: '#FFF';
  font-family: 'roboto-light';
  font-size: 16;
  font-weight: '300';
`;

const Button = ({ text, onPress }) => {
  return (
    <TouchableContainer
      underlayColor="#ffffff"
      activeOpacity={0.3}
      delayLongPress={3800}
      onPress={evt => (onPress ? onPress(evt) : null)}
      onLongPress={evt => (onPress ? onPress(evt) : null)}
    >
      <ButtonContainer>
        <ButtonText>{text}</ButtonText>
      </ButtonContainer>
    </TouchableContainer>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  primary: PropTypes.bool,
};

Button.defaultProps = {
  primary: true,
};

export default Button;
