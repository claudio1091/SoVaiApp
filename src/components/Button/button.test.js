import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Button from './index';

configure({ adapter: new Adapter() });

describe('Testing Button component', () => {
  it('Render as expected', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<Button onPress={onPress} text="Test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render Primary as expected', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<Button primary onPress={onPress} text="Test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Press Button test', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<Button onPress={onPress} text="Test" />);
    wrapper.props().onPress();

    expect(onPress.mock.calls).toHaveLength(1);
  });
});
