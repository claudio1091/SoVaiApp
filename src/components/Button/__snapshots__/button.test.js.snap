// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Render as expected 1`] = `
<Styled(TouchableHighlight)
  activeOpacity={0.3}
  delayLongPress={3800}
  onLongPress={[Function]}
  onPress={[Function]}
  underlayColor="#ffffff"
>
  <Styled(Component)>
    <Styled(Component)>
      Test
    </Styled(Component)>
  </Styled(Component)>
</Styled(TouchableHighlight)>
`;
